tests/rptest/tests/audit_log_test.py

```diff
@@ -1924,84 +1993,44 @@ def test_sr_audit(self):
-
-            return False
-
         records = self.find_matching_record(
-            lambda record: match_api_user("subjects", self.username, self.
-                                          sr_audit_svc_name, record),
+            lambda record: self.match_api_record(record, "subjects"),
             lambda record_count: record_count >= 1, 'sr get api call')
 
-        assert self.aggregate_count(
-            records
-        ) == 1, f'Expected one record found {self.aggregate_count(records)}: {records}'
-
-        def match_authn_user(user, svc_name, result, record):
-            return record['class_uid'] == 3002 and record['dst_endpoint'][
-                'svc_name'] == svc_name and record['user'][
-                    'name'] == user and record['status_id'] == result
+        assert self.aggregate_count(records) == 1, \
+            f'Expected one record found {self.aggregate_count(records)}: {records}'
 
         _ = self.find_matching_record(
-            lambda record: match_authn_user(self.username, self.
-                                            sr_audit_svc_name, 1, record),
+            lambda record: self.match_authn_record(record, StatusID.SUCCESS),
             lambda record_count: record_count == 1, 'authn attempt in sr')
 
     @ok_to_fail_fips
     @cluster(num_nodes=5)
     def test_sr_audit_bad_authn(self):
+        # Not calling self.setup_cluster() here so the user does not exist
         r = get_subjects(self.redpanda.nodes,
                          self.logger,
                          auth=(self.username, self.password))
         assert r.json()['error_code'] == 40101
 
-        def match_authn_user(user, svc_name, result, record):
-            return record['class_uid'] == 3002 and record['dst_endpoint'][
-                'svc_name'] == svc_name and record['user'][
-                    'name'] == user and record['status_id'] == result
-
         _ = self.find_matching_record(
-            lambda record: match_authn_user(self.username, self.
-                                            sr_audit_svc_name, 2, record),
+            lambda record: self.match_authn_record(record, StatusID.FAILURE),
             lambda record_count: record_count > 1, 'authn fail attempt in sr')
 
         try:
             records = self.find_matching_record(
-                lambda record: match_authn_user(self.username, self.
-                                                sr_audit_svc_name, 1, record),
+                lambda record: self.match_authn_record(record, StatusID.SUCCESS
+                                                       ),
                 lambda record_count: record_count > 1,
                 'authn fail attempt in sr')
-            assert f"Should not have found any records but found {self.aggregate_count(records)}: {records}"
+            assert False, f"Should not have found any records but found {self.aggregate_count(records)}: {records}"
         except TimeoutError:
             pass
```

Comment:
suggestion(non-blocking) - `expect_exception(TimeoutError, lambda _: True)` should work here.

Reply:
Yeah, that's much nicer, thanks